arch/lkl/kernel/irq.c

```diff
@@ -50,19 +51,57 @@ static struct irq_info {
 	const char *user;
 } irqs[NR_IRQS];
 
+static bool irqs_enabled;
 
 /**
- * DO NOT run any linux calls (e.g. printk) here as they may race with the
- * existing linux threads.
+ * This function can be called from arbitrary host threads, so do not
+ * issue any Linux calls (e.g. prink) if lkl_cpu_get() was not issued
+ * before.
  */
 int lkl_trigger_irq(int irq)
 {
+	int ret;
+
 	if (!irq || irq > NR_IRQS)
 		return -EINVAL;
 
-	set_irq_status(irq);
+	ret = lkl_cpu_try_get();
```

Comment:
If try_get fails (during syscall) and the delayed irq may be missed because LKL never goes to idle and no syscall is made? Will it cause problem to timers?